[24기 이송현] [Mission 1 완료]Add: 로그인 페이지 기능 완성

vi2920va opened the pull request:
- login 페이지 레이아웃 완성
- login user name, user password 유효성 검사 로직
- main 페이지 레이아웃 진행

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 로그인 버튼 활성화 기능 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- id, pw 한 글자 이상 시 버튼 활성화
- id '@' 포함, pw 5 글자 이상 입력 시 버튼 활성화(예시)

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- css에 불필요한 코드가 많은 같다고 생각합니다😓

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)

- [x]  필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

dipsiiiiiiiiii commented:
기능에 맞게 커밋 잘 나눠주셨네요! 고생하셨습니다!

psy082 commented on students/11th/donghokim/js/login.js:
```diff
@@ -0,0 +1,39 @@
+      return validationFlag.username && validationFlag.password;
+    }
+  }
+})();
```
리뷰해주신 내용을 보고 코드를 변경해보았습니다. 코드를 다시 읽어보니 확실히 가독성이 떨어지고 id와 password의 상태 값을 validationFunction 쪽에서 관리하는 건 적절한 역할 분담이 아닌 것 같다는 것을 느꼈습니다. id와 password의 상태값은 changeButtonStatus에서 그 때 그 때 받아오는 것으로 변경해보았습니다.

psy082 commented on students/11th/donghokim/js/main.js:
```diff
@@ -0,0 +1,63 @@
+const changeStatus = (button, validationCheckFunction, value) => {
+  if (validationCheckFunction(value)) button.disabled = false;
+  else button.disabled = true;
+}
```
validation 함수를 삭제하고 changeStatus에서 바로 상태값을 확인하도록 변경했습니다

psy082 commented on students/11th/donghokim/main.html:
```diff
@@ -0,0 +1,621 @@
+              <li class="info_list"><a class="info_text" href="/explore/locations/">위치</a></li>
+              <li class="info_list"><a class="info_text" href="/directory/profiles/">인기 계정</a></li>
+              <li class="info_list"><a class="info_text" href="/directory/hashtags/">해시태그</a></li>
+              <li class="info_list">
```
스타일 속성인 class에 의미를 부여하려고 하다보니 이렇게 만들게 되었습니다. 스타일로 사용하지 않은 class 속성들을 제거했습니다

psy082 commented:
리뷰해주셔서 감사합니다 ㅎㅎ 크게 도움이 되었습니다 :)

saengmotmi commented:
LGTM :)

vi2920va committed:
Add: 로그인 페이지 기능 완성
- login 페이지 레이아웃 완성
- login user name, user password 유효성 검사 로직
- main 페이지 레이아웃 진행

vi2920va committed:
Add: main 페이지 레이아웃 완성

vi2920va committed:
Modify : 메인 페이지 레이아웃
 - 메인 페이지 레이아웃 수정
 - 메인 페이지에서 댓글 추가 진행

vi2920va committed:
Add: 메인 페이지 댓글 추가 기능
 - 댓글 유효성 검사 로직
 - Enter를 누르면 댓글 추가 기능

vi2920va committed:
Add: ID, PW 유효성 검사
 - id '@' 포함 유효성 검사 로직
 - pw 5글자 이상 검사 로직
